.github/actions/issue-validator/src/repro-link.mjs

```diff
@@ -0,0 +1,74 @@
+/**
+ * This closes an issue if it doesn't have an expected reproduction.
+ */
+
+// @ts-check
+import { setFailed } from '@actions/core'
+import { context, getOctokit } from '@actions/github'
+import { readFileSync } from 'node:fs'
+import { join } from 'node:path'
+
+const __dirname =
+  '/home/runner/work/next.js/next.js/.github/actions/issue-validator/repro-link'
+
+/** @typedef {ReturnType<typeof getOctokit>['rest']} GitHubClient */
+
+async function run() {
+  const token = process.env.GITHUB_TOKEN
+  if (!token) throw new Error('No GITHUB_TOKEN provided')
+
+  const { issue } = context.payload
+  if (!issue) return console.log('Not an issue, exiting')
+
+  const { body: issue_body } = issue
+  if (!issue_body) return console.log('Could not get issue body, exiting')
+
+  const start =
+    '### Link to the code that reproduces this issue or a replay of the bug'
+  const end = '### To Reproduce'
+  const linkRe = new RegExp(`${start}(.*)${end}`, 'is')
+  const match = issue_body.match(linkRe)?.[0].trim()
+
+  if (await hasRepro(match)) {
+    console.log(`Issue #${issue.number} contains a valid reproduction link`)
+    return
+  }
+
+  // A client to load data from GitHub
+  const { rest: client } = getOctokit(token)
+
+  console.log(
+    `Issue #${issue.number} does not contain a valid reproduction link, closing/commenting`
+  )
+
+  // Close the issue
+  await client.issues.update({
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    issue_number: issue.number,
+    state: 'closed',
+  })
+
+  // Comment on the issue
+  await client.issues.createComment({
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    issue_number: issue.number,
+    body: readFileSync(join(__dirname, 'invalid-link.md'), 'utf8'),
+  })
+}
+
+run().catch(setFailed)
+
+/** @param {string|null|undefined} link */
+async function hasRepro(link) {
+  if (!link) return false
+  try {
+    const url = new URL(link)
+    if (!['github.com', 'codesandbox.io'].includes(url.hostname)) return false
```

Comment:
Should we also allow other online IDE solutions such as StackBlitz and Codespaces?

Reply:
Let's start with the ones we already have in the bug report template. We can have a different discussion if we want to expand to allow others